store/src/java/com/zimbra/cs/filter/SieveToSoap.java

```diff
@@ -288,14 +288,20 @@ protected void visitHeaderTest(Node node, VisitPhase phase, RuleProperties props
 
     @Override
     protected void visitHeaderTest(Node node, VisitPhase phase, RuleProperties props,
-            List<String> headers, Sieve.ValueComparison comparison, boolean isCount, String value) {
+            List<String> headers, Sieve.ValueComparison comparison, Sieve.Comparator comparator, boolean isCount, String value) {
         if (phase == VisitPhase.begin) {
             FilterTest.HeaderTest test = addTest(new FilterTest.HeaderTest(), props);
             test.setHeaders(Joiner.on(',').join(headers));
             if (isCount) {
                test.setCountComparison(comparison.toString());
             } else {
                test.setValueComparison(comparison.toString());
+               if (comparator != null) {
+                   test.setValueComparisonComparator(comparator.toString());
+                   if (Sieve.Comparator.ioctet.equals(comparator)) {
```

Comment:
Is "comparator" guaranteed to be lower-case string?  If not, isn't it safer to be "Sieve.Comparator.ioctet.equalsIgnoreCase(comparator)" ?

Reply:
the comparison done here checks  enum reference equality rather than value. So if the comparators are supposed to be case insensitive (couldn't find in rfc), then Sieve.Comparator enum fromString method should do case insensitive comparison. currently it's case sensitive.